RecoMuon/MuonRechitClusterProducer/plugins/RechitClusterProducer.cc

```diff
@@ -0,0 +1,268 @@
+      rParam_(iConfig.getParameter<double>("rParam")),
+      nRechitMin_(iConfig.getParameter<int>("nRechitMin")),
+      nStationThres_(iConfig.getParameter<int>("nStationThres")) {
+  produces<RecHitClusterCollection>();
+}
+
+//
+// member functions
+//
+
+// ------------ method called to produce the data  ------------
+template <typename Trait>
+void RechitClusterProducerT<Trait>::produce(edm::StreamID, edm::Event& ev, const edm::EventSetup& iSetup) const {
+  auto const& geo = iSetup.getData(GeometryToken_);
+  auto const& rechits = ev.get(InputToken_);
+
+  std::vector<fastjet::PseudoJet> fjInput;
+  RecHitRefVector inputs_;
+
+  fastjet::JetDefinition jet_def(fastjet::cambridge_algorithm, rParam_);
+
+  inputs_.clear();
+  fjInput.clear();
+
+  Trait RecHitTrait;
+
+  int recIt = 0;
+  for (auto const& rechit : rechits) {
+    LocalPoint RecHitLocalPosition = rechit.localPosition();
+    auto detid = RecHitTrait.detid(rechit);
+    auto thischamber = geo.chamber(detid);
+    if (thischamber) {
+      GlobalPoint globalPosition = thischamber->toGlobal(RecHitLocalPosition);
+      float x = globalPosition.x();
+      float y = globalPosition.y();
+      float z = globalPosition.z();
+      RechitRef ref = RechitRef(&rechits, recIt);
+      inputs_.push_back(ref);
+      fjInput.push_back(fastjet::PseudoJet(x, y, z, globalPosition.mag()));
+      fjInput.back().set_user_index(recIt);
+    }
+    recIt++;
+  }
+  fastjet::ClusterSequence clus_seq(fjInput, jet_def);
+
+  //keep all the clusters
+  double ptmin = 0.0;
+  std::vector<fastjet::PseudoJet> fjJets = fastjet::sorted_by_pt(clus_seq.inclusive_jets(ptmin));
+
+  auto clusters = std::make_unique<RecHitClusterCollection>();
+  if (!fjJets.empty()) {
+    for (unsigned int ijet = 0; ijet < fjJets.size(); ++ijet) {
+      // get the fastjet jet
+      const fastjet::PseudoJet& fjJet = fjJets[ijet];
+
+      // skip if the cluster has too few rechits
+      if (int(fjJet.constituents().size()) < nRechitMin_)
+        continue;
+      // get the constituents from fastjet
+      RecHitRefVector rechits;
+      for (unsigned int i = 0; i < fjJet.constituents().size(); i++) {
+        auto index = fjJet.constituents()[i].user_index();
+        if (index >= 0 && static_cast<unsigned int>(index) < inputs_.size()) {
+          rechits.push_back(inputs_[index]);
+        }
+      }
+
+      //Derive cluster properties
+      int nStation = 0;
+      int totStation = 0;
+      float avgStation = 0.0;
+      std::map<int, int> station_count_map;
+      for (auto& rechit : rechits) {
+        station_count_map[RecHitTrait.station(*rechit)]++;
+      }
+      //station statistics
+      std::map<int, int>::iterator it;
+      for (auto const& [station, count] : station_count_map) {
+        if (count >= nStationThres_) {
+          nStation++;
+          avgStation += (station) * (count);
+          totStation += (count);
```

Comment:
```suggestion
          avgStation += station * count;
          totStation += count;
```
`)(` seem superfluous